### PR TITLE
Use SortedList to simplify InstructionCollection

### PR DESCRIPTION
**Pull request by BSG-75**
By using `SortedList<int, InstructionBase>`, instruction's position is stored together with instruction itself, which allows fast retrieval of a instruction by its position, and we can avoid to manually calculate branch offsets, so things become a lot simpler xD
(Actually I think we can even remove `InstructionBase.Size` since it's not needed anymore)

### CODE REVIEW

File: src/OpenSage.Game.Tests/Gui/Apt/ActionScript/InstructionParsingTest.cs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    public class InstructionParsingTest
+    {
+        private readonly ITestOutputHelper _output;
+
+        public InstructionParsingTest(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+
```
Comment (feliwir) [inline src/OpenSage.Game.Tests/Gui/Apt/ActionScript/InstructionParsingTest.cs]: Could you just have one empty line here?

File: src/OpenSage.Game.Tests/Gui/Apt/ActionScript/InstructionParsingTest.cs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        public InstructionParsingTest(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        
+
+        // variableName should be parameter2
+        // instruction count should be 19 instead of 20
+        private Stream Get19SimpleInstructions(string variableName, string wrongValue, string rightValue, int afterInstructionPosition)
```
Comment (feliwir) [inline src/OpenSage.Game.Tests/Gui/Apt/ActionScript/InstructionParsingTest.cs]: Did you make up that function or did you take it from some .apt?

File: src/OpenSage.Game/Gui/Apt/ActionScript/InstructionCollection.cs [hunk 3]
```diff
@@ -2,462 +2,503 @@
...
+        private readonly SortedList<int, InstructionBase> _instructions;
+
+        public int Count => _instructions.Count;
+
+        public InstructionCollection(SortedList<int, InstructionBase> instructions)
+        {
+            _instructions = instructions;
+        }
+
+        /* public InstructionCollection(IList<InstructionBase> list) : base(list)
```
Comment (feliwir) [inline src/OpenSage.Game/Gui/Apt/ActionScript/InstructionCollection.cs]: Remove this please if it has no use

File: src/OpenSage.Game.Tests/Gui/Apt/ActionScript/InstructionParsingTest.cs (hunk 2 above)
Comment (BSG-75) [reply src/OpenSage.Game.Tests/Gui/Apt/ActionScript/InstructionParsingTest.cs]: Nah, those instructions were written by hand xD

File: src/OpenSage.Game.Tests/Gui/Apt/ActionScript/InstructionParsingTest.cs (hunk 1 above)
Comment (BSG-75) [reply src/OpenSage.Game.Tests/Gui/Apt/ActionScript/InstructionParsingTest.cs]: All right, will fix them soon

File: src/OpenSage.Game/Gui/Apt/ActionScript/InstructionCollection.cs (hunk 3 above)
Comment (BSG-75) [reply src/OpenSage.Game/Gui/Apt/ActionScript/InstructionCollection.cs]: Fixed

**Pull request merged**